mtags/src/main/scala-2/scala/meta/internal/pc/completions/MatchCaseCompletions.scala

```diff
@@ -68,78 +73,163 @@ trait MatchCaseCompletions { this: MetalsGlobal =>
       case sel => new Parents(sel.pos)
     }
     override def isPrioritized(member: Member): Boolean =
-      member.isInstanceOf[TextEditMember]
+      member match {
+        case m: TextEditMember => !m.filterText.contains("(exhaustive)")
+        case _ => false
+      }
+
     override def contribute: List[Member] = {
-      val result = ListBuffer.empty[Member]
-      val isVisited = mutable.Set.empty[Symbol]
-      def visit(
-          sym: Symbol,
-          name: String,
-          autoImports: List[l.TextEdit]
-      ): Unit = {
-        val fsym = sym.dealiasedSingleType
-        val isValid = !isVisited(sym) &&
-          !isVisited(fsym) &&
-          !parents.isParent(fsym) &&
-          (fsym.isCase ||
-            fsym.hasModuleFlag ||
-            fsym.isInstanceOf[TypeSymbol]) &&
-          parents.isSubClass(fsym, includeReverse = false)
-        def recordVisit(s: Symbol): Unit = {
-          if (s != NoSymbol && !isVisited(s)) {
-            isVisited += s
-            recordVisit(s.moduleClass)
-            recordVisit(s.module)
-            recordVisit(s.dealiased)
+
+      val selectorSym = parents.selector.typeSymbol
+
+      // Special handle case when selector is a tuple or `FunctionN`.
+      if (definitions.isTupleType(parents.selector)) {
+        List(
+          new TextEditMember(
+            "case () =>",
```

Comment:
Does the filter text work with VS Code? There were some issues with non standard characters.

Reply:
I haven't encountered any issues, do you have some specific test in mind to make sure?

Reply:
if it works then it's fine. It's just in some situations VS Code wouldn't show certain completions if the contained some non alphanumeric characters. Might be an issue with string interpolation only though.